Week-6/6.1/assignment6.1.js

```diff
@@ -0,0 +1,47 @@
+// returns the sum of the elements of an array.
+function sumOfArray(arr) {
+    return arr.reduce((partialSum, a) => partialSum + a, 0);
+}
+
+// this is used to maximum sum possible of a contiguous subarray of the passed array
+function maxSumOfSubArray(arr) {
+    let arrLen = arr.length;
+    let maxSum = 0;
+
+    // easy conditions if array length is less than or equal to 2
+    if(arrLen <= 0) {
+        maxSum = -1;
+    } else if(arrLen == 1) {
+        maxSum = arr[0];
+    } else if(arrLen == 2) {
+        maxSum = Math.max(arr[0], arr[1]);
+    } else {
+        // initiate the minimum and maximum possible subarray lengths
+        let MinLen = 2;
+        let MaxLen = arrLen - 1;
+
+        let lennie = MinLen;
+        while(lennie <= MaxLen) {
+            for(let i=0 ; i<arrLen-lennie+1 ; i++) {
+                let newArr = [];
+                for (let el = i ; el<i+lennie ; el++) {
+                    newArr.push(arr[el]);
+                }
+
+                maxSum = Math.max(maxSum, sumOfArray(newArr));
```

Comment:
@mr-rego FYI, I am just suggesting you a method to solve this problem. Learn Kadane's algorithm where the same can be achieved in O(n)